### ISSUE TITLE
Print Exception Issue on GameThread

### ISSUE THREAD

**Issue by Frag1337** [Source-Python-Dev-Team/Source.Python#237]
Hey,

I currently have the problem that some exceptions dont print.

Some purpose TypeErrors showed and some havent, but I noticed it doesnt print when it has been called from GameThread.

I remember there was an issue like this long time before, but the search bar couldnt help me to find this.

> Date          : 2018-01-21 05:06:39.738578
OS            : Windows-10-10.0.16299
Game          : csgo
SP version    : 639
Github commit : 8681755bb6f51bc3c46f19983ac0e0f428f9c7be

**Comment by invincibleqc** [comment]
Reproducible code?

**Comment by Frag1337** [comment]
> from listeners.tick import GameThread

def do_post_request():
    not_defined_function()

thread = GameThread(target=do_post_request)
thread.start()

**Comment by invincibleqc** [comment]
Seems like the exception is written directly to `stderr` which is not internally bound to the console on Windows 10 due to [the fix](https://github.com/Source-Python-Dev-Team/Source.Python/blob/master/src/core/sp_python.cpp#L210-L215) for issue #175. And due to that, our `excepthook` is not called and the traceback never written to the console. If I bind `stderr` to the `CONOUT$` channel the exception is correctly printed. You can add `sys.stderr = io.open('CONOUT$', 'wt')` at the top of your code to test it out.

**Comment by Frag1337** [comment]
```
[SP] Loading plugin 'test'...
[SP] Successfully loaded plugin 'test'.
Exception in thread Thread-37:
Traceback (most recent call last):
  File "D:\GameServer\csgo\addons\source-python\Python3\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "D:\GameServer\csgo\addons\source-python\Python3\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "D:\GameServer\csgo\addons\source-python\plugins\test\test.py", line 9, in do_post_request
    not_defined_function()
NameError: name 'not_defined_function' is not defined
```
The traceback is being shown now in the console with the addition on the top.

**Comment by Ayuto** [comment]
>Seems like the exception is written directly to stderr which is not internally bound to the console on Windows 10 due to the fix for issue #175. And due to that, our excepthook is not called and the traceback never written to the console.

Actually, it's not printed, because our sys.excepthook didn't get called. This is also true for servers that didn't reconnect the output streams. However, I will also connect ``sys.stderr`` to ``CONOUT$`` as ``CONERR$`` obviously has no effect.
